plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/MarquezServer.java

```diff
@@ -0,0 +1,122 @@
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.Closeable;
+import java.net.URI;
+import java.time.Duration;
+
+import static java.lang.String.format;
+
+public class MarquezServer
+        implements Closeable
+{
+    private static final String MARQUEZ_CONFIG;
+    private static final int MARQUEZ_PORT;
+    private static final int MARQUEZ_ADMIN_PORT;
+    private static final String MARQUEZ_HEALTCHECK_API;
+    private static final String POSTGRES_HOST;
+    private static final String POSTGRES_DB;
+    private static final int POSTGRES_PORT;
+    private static final String POSTGRES_USER;
+    private static final String POSTGRES_PASSWORD;
+    private static final String DEFAULT_VERSION;
+
+    private Network network;
+
+    static {
+        MARQUEZ_CONFIG = "/opt/marquez/marquez.test.yaml";
+        MARQUEZ_PORT = 7000;
+        MARQUEZ_ADMIN_PORT = 7001;
+        MARQUEZ_HEALTCHECK_API = "/ping";
+        POSTGRES_HOST = "postgres";
+        POSTGRES_DB = "marquez";
+        POSTGRES_PORT = 5432;
+        POSTGRES_USER = "marquez";
+        POSTGRES_PASSWORD = "marquez";
+        DEFAULT_VERSION = "0.46.0";
+    }
+
+    private final GenericContainer<?> dockerContainerAPI;
+    private final GenericContainer<?> dockerContainerPostgres;
+
+    public MarquezServer()
+    {
+        this(DEFAULT_VERSION);
+    }
+
+    public MarquezServer(String version)
+    {
+        network = Network.newNetwork();
+        this.dockerContainerPostgres = new GenericContainer<>("postgres:14")
+                .withNetwork(network)
+                .withNetworkAliases(POSTGRES_HOST)
+                .withEnv("POSTGRES_USER", POSTGRES_USER)
+                .withEnv("POSTGRES_PASSWORD", POSTGRES_PASSWORD)
+                .withEnv("POSTGRES_DB", POSTGRES_DB)
+                .withStartupTimeout(Duration.ofSeconds(360));
+
+        this.dockerContainerPostgres.start();
+
+        this.dockerContainerAPI = new GenericContainer<>("marquezproject/marquez:" + version)
+                .withNetwork(network)
+                .withExposedPorts(MARQUEZ_PORT, MARQUEZ_ADMIN_PORT)
+                .withEnv("MARQUEZ_PORT", String.valueOf(MARQUEZ_PORT))
+                .withEnv("MARQUEZ_ADMIN_PORT", String.valueOf(MARQUEZ_ADMIN_PORT))
+                .withEnv("POSTGRES_URL", getPostgresUri())
+                .withEnv("POSTGRES_USER", "marquez")
+                .withEnv("POSTGRES_PASSWORD", "marquez")
+                .withEnv("MARQUEZ_CONFIG", MARQUEZ_CONFIG)
+                .withCopyFileToContainer(
+                        MountableFile.forClasspathResource("marquez.yaml"),
+                        MARQUEZ_CONFIG)
+                .withCommand("/opt/marquez/wait-for-it.sh", format("%s:%s", POSTGRES_HOST, POSTGRES_PORT), "--", "./entrypoint.sh")
```

Comment:
Instead can we `dependsOn` which ensures this container would be started only after postgres

Reply:
done!